client/src/test/java/se/fortnox/reactivewizard/client/HttpClientTest.java

```diff
@@ -274,6 +273,27 @@ public void shouldNotSometimesDropItems() {
         });
     }
 
+    @Test
+    public void shouldReuseJaxRsMetaObjectsToAvoidReflection() throws URISyntaxException, NoSuchMethodException {
+
+        List<JaxRsMeta> jaxRsMetas = new ArrayList<>();
+        HttpClient httpClient = new HttpClient(new HttpClientConfig("localhost")) {
+            @Override
+            protected JaxRsMeta getJaxRsMeta(Method method) {
+                JaxRsMeta jaxRsMeta = super.getJaxRsMeta(method);
+                jaxRsMetas.add(jaxRsMeta);
+                return jaxRsMeta;
+            }
+        };
+
+        Method getHello = TestResource.class.getMethod("getHello");
+
+        httpClient.createRequest(getHello,new Object[0]);
+        httpClient.createRequest(getHello,new Object[0]);
+
+        assertThat(jaxRsMetas.get(0)).isSameAs(jaxRsMetas.get(1));
```

Comment:
A test with another method would be nice, just to show it doesn't cache just one object. :)